pkg/client/elastic.go

```diff
@@ -107,15 +108,16 @@ func (c *ESClient) Add(record record.Record, index string, rtype string) {
 }
 
 // Promote will add the given index to the primary alias. If there is an
-// existing index matching the prefix linked to the primary alias it will
-// be removed from the alias. This action is atomic.
-func (c ESClient) Promote(index string, prefix string) error {
-	svc := c.client.Alias().Add(index, primary)
-	current, err := c.Current(prefix)
+// existing index with the same prefix as the promoted index and linked to the
+// primary alias, it will be removed from the alias. This action is atomic.
+func (c ESClient) Promote(index string) error {
+  svc := c.client.Alias().Add(index, primary)
+  prefix := strings.Split(index, "-")[0]
```

Comment:
Let's document the expectation of the prefix always having a trailing `-`. I sent you details on an issue I ran into on an index that was `prefix_` instead of `prefix-`.

Reply:
Resolved by enforcing the index naming pattern during ingest. We'll still need to manually resolve this in staging/prod, but it shouldn't be an issue going forward.